.github/workflows/builds.yml

```diff
@@ -0,0 +1,155 @@
+      with:
+        hypre-archive: ${{ env.HYPRE_ARCHIVE }}
+        hypre-dir: ${{ env.HYPRE_TOP_DIR }}
+
+    # Get Metis through cache, or build it.
+    # Install will only run on cache miss.
+    - name: cache metis
+      id: metis-cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.METIS_TOP_DIR }}
+        key: ${{ runner.os }}-build-${{ env.METIS_TOP_DIR }}-v2
+
+    - name: build metis
+      if: steps.metis-cache.outputs.cache-hit != 'true'
+      uses: mfem/mfem-actions/build-metis@master
+      with:
+        metis-archive: ${{ env.METIS_ARCHIVE }}
+        metis-dir: ${{ env.METIS_TOP_DIR }}
+
+    # make generic links to libraries for MFEM install
+    # Those links are already created by build-mfem action, but not when the cache is used...
+    - name: configure links
+      run: |
+        echo "Hypre symlink:"
+        ln -s $HYPRE_TOP_DIR hypre;
+        echo "Metis symlink:"
+        ln -s $METIS_TOP_DIR metis-4.0;
+
+    - name: MFEM ${{ env.MFEM_BRANCH }} commit
+      run: |
+        echo "MFEM_COMMIT=$(git ls-remote --heads https://github.com/mfem/mfem.git ${{ env.MFEM_BRANCH }} | awk '{print $1;}')" >> $GITHUB_ENV
+
+    # Get MFEM through cache, or build it.
+    # Install will only run on cache miss.
+    - name: cache mfem
+      id: mfem-cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.MFEM_TOP_DIR }}
+        key: ${{ runner.os }}-build-${{ env.MFEM_TOP_DIR }}-${{ env.MFEM_COMMIT }}-${{ matrix.target }}-v3
+
+    # We are using the defaults of the MFEM action here, which is to use master
+    # branch. There is an implicit assumption here that mfem master hasn't
+    # changed since the 'MFEM master commit' step.
+    # Also, if we apply to MFEM build the same target as GLVis. This may be
+    # superfluous.
+    - name: build mfem
+      if: steps.mfem-cache.outputs.cache-hit != 'true'
+      uses: mfem/mfem-actions/build-mfem@master
+      with:
+        os: ${{ matrix.os }}
+        target: ${{ matrix.target }}
+        hypre-dir: ${{ env.HYPRE_TOP_DIR }}
+        metis-dir: ${{ env.METIS_TOP_DIR }}
+        mfem-dir: ${{ env.MFEM_TOP_DIR }}
+        mfem-branch: ${{ env.MFEM_BRANCH }}
+
+    # Install GlVis dependencies with package manager
+    - name: get deps (Linux)
+      if: matrix.os == 'ubuntu-18.04'
+      run: |
+        sudo apt-get install libfontconfig1-dev libfreetype6-dev libsdl2-dev libglew-dev libglm-dev libpng-dev
+
+    - name: get deps (MacOS)
+      if: matrix.os == 'macos-10.15'
+      run: |
+        export HOMEBREW_NO_INSTALL_CLEANUP=1
+        brew install fontconfig freetype sdl2 glew glm libpng
+
+    - name: checkout GLVis
+      uses: actions/checkout@v2
+      with:
+        path: glvis
+
+    - name: build GLVis
+      run: |
+        glvis_target=""
+        [[ ${{ matrix.target }} == "optim" ]] && glvis_target="opt";
+        [[ ${{ matrix.target }} == "debug" ]] && glvis_target="debug";
+        cd glvis && make ${glvis_target} -j
```

Comment:
Would it be good to also include a build test for the CMake build system?

Reply:
I think it will definitely be good to add that

Reply:
@kanye-quest does building GLVis with CMake requires building MFEM with CMake first?
I see in the documentation that the example points to MFEMConfig.cmake.

Reply:
Yes, GLVis' CMake build expects MFEM build with CMake.

Reply:
Done